### ISSUE TITLE
<select> with multiple formats params in such a way that some interpreters ignore all but last

### ISSUE THREAD

**Issue by kevans91** [jquery-form/form#417]
When a `select` element with the `multiple` attribute specified is present in a form, only the last value selected is sent along when the form is submitted.

My work-around for this specific issue, which might be less than ideal, was the following:

```
992c992
<                 a.push({name: n, value: v[j]});

---
>                 a.push({name: n + "[]", value: v[j]});
```

_Diff made using version 3.50.0-2014.02.05 for reference_

The problem seems to occur because jquery.param does not bother with whether or not there exists multiple values for a given key, it instead just takes what it's given and encodes it as a query string.

The alternative would be to pass an array as the value for these to jquery.param instead of multiple objects with the same "name" value, but that returns a comma-separated list of the values instead.

---

Additionally, this behavior with jquery.param could also be a problem further down on line 1002, which is a similar scenario with a similar fix:

``` .
                     a.push({name: n, value: files[j], type: el.type});
```

Perhaps related to #403 in some respect.

**Comment by kevans91** [comment]
Aware that this issue only really applies when the script processing the request doesn't parse multiple similar key assignments in the query string as an array and catering to PHP and a small handful of others isn't always a great idea, but it seems like this would be the most intuitive way for this to be handled even for languages that can handle this type of string properly or don't process the query string at all.

**Comment by kevindb** [comment]
@kevans91, the commit you referenced, [fbf51ba](https://github.com/kevans91/form/commit/fbf51ba2e2266b5463c1e3b648ed9055f043c1ec), looks like it will _always_ add brackets `[]` to the field name, rather than _only_ when the `select` has the `multiple` attribute set.
Three years later, we, as a society, still don't have a standardized way of representing arrays in query strings. However, this answer on SO offers a pretty good solution:
http://stackoverflow.com/a/9547490/1480959

TLDR: Instead of naming your field `cars`, name it `cars[]`. That will allow the brackets to be in the field name so your PHP script recognizes the multiple values as an array.

**Comment by kevans91** [comment]
Yeah, so, I believe I had tried that and ran into a different problem entirely. I don't recall, though -- as you've mentioned, it's been three years, I don't recall what I was even using this for at the time and I've since moved on to different projects and actually no longer use jQuery form at all. =(
